src/credentials/sso/diskCache.ts

```diff
@@ -4,80 +4,83 @@
         if (this.registrationExists(ssoRegion)) {
-            fs.unlinkSync(join(this.cacheDir, `aws-toolkit-vscode-client-id-${ssoRegion}.json`))
+            fs.unlinkSync(joinPath(this.cacheDir, this.registrationFilename(ssoRegion)))
         }
     }
-    loadAccessToken(ssoUrl: string) {
+    public loadAccessToken(ssoUrl: string): SsoAccessToken | undefined {
         if (this.tokenExists(ssoUrl)) {
-            return JSON.parse(fs.readFileSync(join(this.cacheDir, this.accessTokenCache(ssoUrl))).toString())
+            return JSON.parse(fs.readFileSync(joinPath(this.cacheDir, this.accessTokenCache(ssoUrl))).toString())
         }
+        return undefined
     }
 
-    saveAccessToken(ssoUrl: string, accessToken: AccessToken) {
+    public saveAccessToken(ssoUrl: string, accessToken: SsoAccessToken): void {
         const fileName = this.accessTokenCache(ssoUrl)
-        fs.writeFileSync(join(this.cacheDir, `${fileName}.json`), JSON.stringify(accessToken), { mode: 0o600 })
+        fs.writeFileSync(joinPath(this.cacheDir, `${fileName}.json`), JSON.stringify(accessToken), { mode: 0o600 })
     }
 
-    invalidateAccessToken(ssoUrl: string) {
+    public invalidateAccessToken(ssoUrl: string): void {
         if (this.tokenExists(ssoUrl)) {
-            fs.unlinkSync(join(this.cacheDir, this.accessTokenCache(ssoUrl)))
+            fs.unlinkSync(joinPath(this.cacheDir, this.accessTokenCache(ssoUrl)))
         }
     }
 
     private accessTokenCache(ssoUrl: string): string {
-        // According to the SEP 'SSO Login Token Flow' the access token should be cached as
+        // According to the Spec: 'SSO Login Token Flow' the access token should be cached as
         // the SHA1 hash of the bytes of the UTF-8 encoded startUrl value with .json appended to the end.
         const encoded = encodeURI(ssoUrl)
-        let shasum = crypto.createHash('sha1')
-        shasum.update(encoded)
-        return shasum.digest('hex')
+        return getSHA1StringHash(encoded)
     }
 
     private registrationExists(ssoRegion: string): boolean {
-        return fs.existsSync(join(this.cacheDir, `aws-toolkit-vscode-client-id-${ssoRegion}.json`))
+        return fs.existsSync(joinPath(this.cacheDir, this.registrationFilename(ssoRegion)))
     }
 
     private tokenExists(ssoUrl: string): boolean {
-        return fs.existsSync(join(this.cacheDir, this.accessTokenCache(ssoUrl)))
+        return fs.existsSync(joinPath(this.cacheDir, this.accessTokenCache(ssoUrl)))
     }
 
-    // Treat the token or client registration as expired if within 15 minutes of expiration.
     private isNotExpired(token: any): boolean {
-        return Date.parse(token.expiresAt) - 900000 > Date.now()
+        return Date.parse(token.expiresAt) - this.TIME_ADDED_BEFORE_EXPIRY > Date.now()
+    }
+
+    private registrationFilename(ssoRegion: string): string {
+        return `aws-toolkit-vscode-client-id-${ssoRegion}.json`
```

Comment:
Is there any reason not to move the `join(this.cacheDir, x)` logic into here as well?

Reply:
Great point, this cleans it up a bit further.